buildSrc/src/main/java/com/google/firebase/gradle/plugins/semver/ApiDiffer.kt

```diff
@@ -0,0 +1,182 @@
+
+abstract class ApiDiffer : DefaultTask() {
+  @get:Input abstract val aarPath: Property<File>
+
+  @get:Input abstract val artifactId: Property<String>
+
+  @get:Input abstract val groupId: Property<String>
+
+  @get:Input abstract val version: Property<String>
+
+  private val CLASS_EXTENSION = ".class"
+
+  @TaskAction
+  fun run() {
+    val gMavenHelper = GmavenHelper(groupId.get(), artifactId.get())
+    val previousVersion = gMavenHelper.getLatestReleasedVersion().split(".")
+    if (previousVersion.size != 3) {
+      return
+    }
+    val versionlist = version.get().split(".")
+    var curVersionDelta: VersionDelta
+    if (versionlist[0] > previousVersion[0]) {
+      curVersionDelta = VersionDelta.MAJOR
+    } else if (versionlist[1] > previousVersion[1]) {
+      curVersionDelta = VersionDelta.MINOR
+    } else {
+      curVersionDelta = VersionDelta.PATCH
+    }
+    val currentAarClassJar = aarPath.get().absolutePath.replace(".aar", "")
+    val afterJar = readApi(aarPath.get(), currentAarClassJar)
+    val lastAarPath = gMavenHelper.getAarFileForVersion(gMavenHelper.getLatestReleasedVersion())
+
+    val previousAarPath: String = aarPath.get().absolutePath.replace(".aar", "/old.aar")
+    try {
+      BufferedInputStream(URL(lastAarPath).openStream()).use { `in` ->
+        FileOutputStream(previousAarPath).use { fileOutputStream ->
+          val dataBuffer = ByteArray(1024)
+          var bytesRead: Int
+          while (`in`.read(dataBuffer, 0, 1024).also { bytesRead = it } != -1) {
+            fileOutputStream.write(dataBuffer, 0, bytesRead)
+          }
+        }
+      }
+    } catch (e: IOException) {
+      // handle exception
+    }
+
+    val previousAarClassJar = previousAarPath.replace(".aar", "")
+    val beforeJar = readApi(File(previousAarPath), previousAarClassJar)
+    val apiDeltas = mutableListOf<Delta>()
+    val classKeys = afterJar.keys union beforeJar.keys
+    classKeys.forEach {
+      val afterClass = afterJar.get(it)
+      val beforeClass = beforeJar.get(it)
+      val allDeltas = DeltaType.values()
+      allDeltas.forEach { apiDeltas.addAll(it.getViolations(beforeClass, afterClass)) }
+    }
+    var deltaViolations: List<Delta> = mutableListOf()
+    if (curVersionDelta == VersionDelta.MINOR) {
+      deltaViolations = apiDeltas.filter { it.versionDelta == VersionDelta.MAJOR }
+    } else if (curVersionDelta == VersionDelta.PATCH) {
+      deltaViolations = apiDeltas
+    }
+    if (apiDeltas.size != 0) {
+      println(
+        "Here is a list of all the minor/major version bump changes which are made since the last release"
+      )
+      apiDeltas.forEach { println("[${it.versionDelta}] ${it.description}") }
+    }
+    if (deltaViolations.size != 0) {
+      var outputString =
+        "Here is a list of all the violations which needs to be fixed before we could release.\n"
+      deltaViolations.forEach { outputString += "[${it.versionDelta}] ${it.description}\n" }
```

Comment:
```suggestion
      val outputString =
              deltaViolations.joinToString(
        prefix =
          "Here is a list of all the minor/major version bump changes which are made since the last release",
        separator = "\n"
      ) {
        "[${it.versionDelta}] ${it.description}"
      }
```